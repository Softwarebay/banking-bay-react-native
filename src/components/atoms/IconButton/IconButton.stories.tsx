import { action } from '@storybook/addon-actions';
import { color, object, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { colors } from 'theme';

import { CenterStory } from 'utils/storybook/CenterStory';

import { IconButton } from './IconButton';

const btnStylesKnob = {
	width: 60,
	height: 60,
	borderRadius: 100,
};

storiesOf('Atoms/IconButton', module)
	.addDecorator((getStory) => <CenterStory center>{getStory()}</CenterStory>)
	.add('Facebook', () => (
		<IconButton
			btnStyle={object('btnStyle', {
				...btnStylesKnob,
				backgroundColor: '#3B5999',
			})}
			iconColor={color('iconColor', colors.white)}
			iconName="facebook"
			testID={text('testID', 'IconButton-test-id')}
			onPress={action('onPress')}
		/>
	))
	.add('Twitter', () => (
		<IconButton
			btnStyle={object('btnStyle', {
				...btnStylesKnob,
				backgroundColor: '#55ACEE',
			})}
			iconColor={color('iconColor', colors.white)}
			iconName="twitter"
			testID={text('testID', 'IconButton-test-id')}
			onPress={action('onPress')}
		/>
	))
	.add('Google', () => (
		<IconButton
			btnStyle={object('btnStyle', {
				...btnStylesKnob,
				backgroundColor: '#DD4B39',
			})}
			iconColor={color('iconColor', colors.white)}
			iconName="google-plus"
			testID={text('testID', 'IconButton-test-id')}
			onPress={action('onPress')}
		/>
	));
