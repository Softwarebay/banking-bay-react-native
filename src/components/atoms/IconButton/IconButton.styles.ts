import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	btn: {
		alignItems: 'center',
		borderRadius: 100,
		height: 60,
		justifyContent: 'center',
		width: 60,
	},
});
