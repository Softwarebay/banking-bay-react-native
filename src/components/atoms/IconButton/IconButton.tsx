import { Pressable, StyleProp, ViewStyle } from 'react-native';

import { Icon, Props as IconProps } from '../Icon/Icon';

import { styles } from './IconButton.styles';

interface Props {
	iconName: IconProps['name'];
	iconColor: IconProps['color'];
	onPress: () => void;
	testID: string;
	btnStyle?: StyleProp<ViewStyle>;
}

export const IconButton = ({
	iconName,
	iconColor,
	btnStyle,
	testID,
	onPress,
}: Props) => {
	return (
		<Pressable
			hitSlop={20}
			style={({ pressed }) => [styles.btn, btnStyle, pressed && styles.pressed]}
			testID={testID}
			onPress={onPress}
		>
			<Icon color={iconColor} name={iconName} />
		</Pressable>
	);
};
