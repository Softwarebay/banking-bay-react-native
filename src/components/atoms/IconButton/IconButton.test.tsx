import {
	fireEvent,
	render,
	screen,
	within,
} from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { IconButton } from './IconButton';

import type { Props } from './IconButton';

describe('IconButton', () => {
	const mockOnPress = jest.fn<Props['onPress'], []>();
	const props: Props = {
		onPress: mockOnPress,
		iconColor: 'white',
		iconName: 'facebook',
		iconSize: 50,
		testID: 'facebook-icon-button-test-id',
		btnStyle: {
			backgroundColor: 'blue',
		},
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<IconButton {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render button with nested icon', () => {
		render(<IconButton {...props} />);

		const iconBtn = screen.getByTestId('facebook-icon-button-test-id');

		const icon = within(iconBtn).getByTestId('facebook-icon-test-id');

		expect(icon.props).toHaveProperty('width', 50);
		expect(icon.props).toHaveProperty('height', 50);
		expect(icon.props).toHaveProperty('color', 'white');
	});

	it('should apply additional button styles', () => {
		render(<IconButton {...props} />);

		const iconBtn = screen.getByTestId('facebook-icon-button-test-id');

		expect(iconBtn).toHaveStyle({ backgroundColor: 'blue' });
	});

	it('should handle press events', () => {
		render(<IconButton {...props} />);

		const iconBtn = screen.getByTestId('facebook-icon-button-test-id');

		fireEvent.press(iconBtn);

		expect(mockOnPress).toHaveBeenCalledTimes(1);
	});
});
