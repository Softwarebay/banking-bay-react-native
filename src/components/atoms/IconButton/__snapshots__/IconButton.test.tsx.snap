// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={20}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "blue",
      },
      false,
    ]
  }
  testID="facebook-icon-button-test-id"
>
  <SvgMock
    color="white"
    height={50}
    testID="facebook-icon-test-id"
    width={50}
  />
</View>
`;
